diff/diff_test.go

```diff
@@ -195,29 +195,61 @@ metadata:
 `,
 		}}
 
+	specReleaseRenamed := map[string]*manifest.MappingResult{
+		"default, nginx-renamed, Deployment (apps)": {
+
+			Name: "default, nginx-renamed, Deployment (apps)",
+			Kind: "Deployment",
+			Content: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-renamed
+`,
+		}}
+
 	t.Run("OnChange", func(t *testing.T) {
 
 		var buf1 bytes.Buffer
 
-		if changesSeen := Manifests(specBeta, specRelease, []string{}, true, 10, "diff", false, &buf1); !changesSeen {
+		if changesSeen := Manifests(specBeta, specRelease, []string{}, true, 10, 0, "diff", false, &buf1); !changesSeen {
 			t.Error("Unexpected return value from Manifests: Expected the return value to be `true` to indicate that it has seen any change(s), but was `false`")
 		}
 
 		require.Equal(t, `default, nginx, Deployment (apps) has changed:
-  
+
 - apiVersion: apps/v1beta1
 + apiVersion: apps/v1
   kind: Deployment
   metadata:
     name: nginx
-  
+
+`, buf1.String())
+	})
+
+	t.Run("OnChangeRename", func(t *testing.T) {
+
+		var buf1 bytes.Buffer
+
+		if changesSeen := Manifests(specRelease, specReleaseRenamed, []string{}, true, 10, 0.34, "diff", false, &buf1); !changesSeen {
+			t.Error("Unexpected return value from Manifests: Expected the return value to be `true` to indicate that it has seen any change(s), but was `false`")
+		}
+
+		require.Equal(t, `default, nginx, Deployment (apps) has changed:
+
+  apiVersion: apps/v1
+  kind: Deployment
+  metadata:
+-   name: nginx
++   name: nginx-renamed
+
```

Comment:
Would you mind adding one more test for the case that there's change in `metadata.name` plus there're one update, one deletion, and one insertion under `spec`?
I think that would make this feature extremely hard to get regression in the future!

Reply:
Sure, I'll add them.